app/apollo/init.local.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2020 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const cors = require('cors');
+const { SECRET } = require('./models/const');
+
+const initApp = (app, models, logger) => {
+  logger.info('Initialize apollo application for local auth');
+  app.use(cors());
+  app.use(function errorHandler(err, req, res, next) {
+    if (err) {
+      if (req.log && req.log.error) req.log.error(err);
+      else logger.error(err);
+      if (!res.headersSent) {
+        const statusCode = err.statusCode || 500;
+        return res.status(statusCode).send();
+      }
+      return next(err);
+    }
+    return next();
+  });
+};
+
+const buildApolloContext = async ({ models, req, res, connection, logger }) => {
+  if (connection) {
+    logger.debug({ connection, req, res }, 'context websocket connection is');
```

Comment:
attach a logger to the request with express middleware, then always use `req.log` for logging

Reply:
For both WS connections and normal HTTP requests, we passed logger into context at line 43 and 53, and use the logger in revolvers.

Reply:
fixed in graphql branch, add `req_id` to graphql context, which is then used by logger to print debug information with the req_id. see [here](https://github.com/razee-io/Razeedash-api/blob/graphql/app/apollo/index.js#L54) for setting and [here](https://github.com/razee-io/Razeedash-api/blob/graphql/app/apollo/resolvers/resource.js#L43) for example how to use.